### PR TITLE
Registry image

### PR DESCRIPTION
**Pull request by spgreenberg**
## Changes proposed in this pull request:

- the registry-image concourse type does not want the https:// in the host setting. Therefore, this has been removed from credhub and added to the docker-image resource.
- Adds a test of both the registry-image and docker-image resource types in the staging smoke tests.

## security considerations

None. This does not affect the registry or registry access rules.

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -217,7 +216,6 @@ resources:
   type: git
   source:
     uri: ((concourse-config-git-url))
-    branch: ((concourse-config-git-branch))
```
Comment (spgreenberg) [inline ci/pipeline.yml:220]: We can omit the branch and concourse will use the default for that repo

File: ci/smoke-test.sh [hunk 2]
```diff
@@ -26,7 +27,28 @@ run:
...
+  type: registry-image
+  source:
+    repository: busybox
+
+run:
+  path: echo
+  args: ["smoke"]
+EOF
+
+output=$(./fly --target ci execute --config ./registry-image-config.yml)
```
Comment (spgreenberg) [inline ci/smoke-test.sh:51]: This block has the smoke test run two tasks, one that uses the docker-image resource type and the other registry-image. This should ensure that the registry mirror settings for both resource types are working correctly.

File: operations/base-resource-defaults.yml [hunk 3]
```diff
@@ -5,4 +5,4 @@
       registry_mirror:
         host: ((docker-registry-mirror))
     docker-image:
-      registry_mirror: ((docker-registry-mirror))
+      registry_mirror: https://((docker-registry-mirror))
```
Comment (spgreenberg) [inline operations/base-resource-defaults.yml:8]: The docker-image wants the https:// and we removed it from credhub because of the line above.

**Review by ChrisMcGowan — APPROVED** [review]
LGTM

**Pull request merged**